index.js

```diff
@@ -87,122 +33,235 @@ function busboy (options) {
   }
 }
 
-function fastifyMultipart (fastify, options, done) {
-  if (options.addToBody === true) {
-    if (typeof options.sharedSchemaId === 'string') {
-      fastify.addSchema({
-        $id: options.sharedSchemaId,
-        type: 'object',
-        properties: {
-          encoding: { type: 'string' },
-          filename: { type: 'string' },
-          limit: { type: 'boolean' },
-          mimetype: { type: 'string' }
-        }
-      })
-    }
-
-    fastify.addHook('preValidation', function (req, reply, next) {
-      attachToBody(options, req, reply, next)
-    })
-  }
+function fastifyMultipart (fastify, options = {}, done) {
   fastify.addContentTypeParser('multipart', setMultipart)
-
-  fastify.decorateRequest('multipart', multipart)
+  fastify.decorateRequest('handleMultipart', handleMultipart)
+  fastify.decorateRequest('multipart', getMultipartIterator)
```

Comment:
If we leave this prop with as the old interface `const bb = req.multipart(handler, onEnd)` could be a solution @jsumners ?

Reply:
I think so. But it might be difficult to support both interfaces.

Reply:
The new interface provides the same flexibility: Consuming streams. On top of that, we can work with files directly and support async/await.

IMO it's not worth supporting both interfaces or what do you see that I don't see?